assgnment2/Functions.dart

```diff
@@ -0,0 +1,256 @@
+import 'dart:convert';
+import 'util.dart';
+import 'user.dart';
+import 'dart:io';
+
+bool validateUserChoice(int userChoice) {
+  if (userChoice < 1 || userChoice > 5) {
+    print('please enter an integer ranging 1-5');
+    return false;
+  }
+  return true;
+}
+
+int takeUserChoice() {
+  int validChoice = Util.validateInteger();
+
+  while (validateUserChoice(validChoice) == false) {
+    printOptions();
+    validChoice = Util.validateInteger();
+  }
+  return validChoice;
+}
+
+void printOptions() {
+  print('Enter the as per options below');
+  print('1:Add user details');
+  print('2:Display the existing user detaiils');
+  print('3:Delete the user details');
+  print('4:Save the user details');
+  print('5:Exit');
+}
+
+int validRollNumber() {
+  int isValid = Util.validateInteger();
+
+  while (User.regieteredRollNumber.contains(isValid)) {
+    print('please enter a unique roll number');
+    isValid = Util.validateInteger();
+  }
+  return isValid;
+}
+
+List<String> readCources() {
+  Set<String> courcesOffered = {"A", "B", "C", "D", "E", "F"};
+  Set<String> currentCources = {};
+
+  print('Cources offered are A,B,C,D,E,F');
+  print("enter the unique cources offered");
+  while (currentCources.length < 4) {
+    String cource = stdin.readLineSync()!;
+    if (courcesOffered.contains(cource)) currentCources.add(cource);
+  }
+
+  print('choosen cources are');
+  List<String> courceChosen = [];
+
+  for (var curr in currentCources) {
+    print(curr);
+    courceChosen.add(curr);
+  }
+
+  return courceChosen;
+}
+
+int validUserAge() {
+  bool isValid = false;
+  int userAge = 0;
+  while (!isValid) {
+    print('enter the Valid  Age of user');
+    userAge = Util.validateInteger();
+    if (userAge > 0 && userAge < 100) isValid = true;
```

Comment:
No need to add upper bound of 100 here. If you still want to add, make it slightly larger to cater for all possible values